src/plugins/intel_gpu/src/graph/impls/ocl/kernels_cache.cpp

```diff
@@ -154,7 +154,11 @@ void kernels_cache::get_program_source(const kernels_code& kernels_source_code,
             std::string options = kernel_string->options;
             bool batch_compilation = kernel_string->batch_compilation;
 
-            if (batch_compilation) {
+            bool is_cm = options.find("-cmc") != std::string::npos;
```

Comment:
Is it the only way to check that CM compiler is invoked?

Reply:
Yes, OpenCL runtime will use CM compiler only if the `-cmc` is present in options provided to clBuildProgram. This is the most straightforward way to check this. Alternatively, we can add `is_cm` flag to `kernel_string` (`src/plugins/intel_gpu/include/intel_gpu/runtime/kernel_args.hpp`) and set it for CM primitives during creation.

Reply:
Usually search in a string being inside a loop is not the most effective way from performance perspective. But overall impact for the app should not be visible I believe. So leaving this for OV guys to suggest.

Reply:
I think we may add enum like `KernelLanguage {OCLC, CM}` and make it a part of `KernelCode` structure